es-app/src/SystemScreenSaver.cpp

```diff
@@ -301,8 +301,8 @@ void SystemScreenSaver::pickGameListNode(unsigned long index, const char *nodeNa
 		pugi::xml_document doc;
 		pugi::xml_node root;
 
-		// We only want images from game systems
-		if (!(*it)->isGameSystem())
+		// We only want nodes from game systems that are not collections
+		if (!(*it)->isGameSystem() || (*it)->isCollection())
```

Comment:
I think that's wise, but in that case isn't the logic reversed? Should it not be

!((*it)->isGameSystem() || (*it)->isCollection())

(missing parentheses around both tests)

That being said, it's probably a non-issue, as collections do not (or should not :P ) have a gamelist path.

But it's certainly more readable, well done.